src/cache.ts

```diff
@@ -139,12 +139,13 @@ export class CMakeCache {
       if (line.startsWith('//')) {
         docs_acc += /^\/\/(.*)/.exec(line)![1] + ' ';
       } else {
-        const match = /^(.*?):(.*?)=(.*)/.exec(line);
+        const match = /^("(.*?)"|(.*?)):([^:]*?)=(.*)/.exec(line);
         if (!match) {
           rollbar.error(localize('failed.to.read.line.from.cmake.cache.file', 'Failed to read a line from a CMake cache file {0}', line));
           continue;
         }
-        const [, name, typename, valuestr] = match;
+        const filtered = match.filter(element => element != undefined);
```

Comment:
tslint is complaining about this line.  I believe you can just do:
```
const filtered = match.filter(element => element);
```

Reply:
Let me try your solution.  It's cleaner than what I came up with.

Reply:
Looks like it works! Thanks for the suggestion.

Reply:
Did you push it?  It looks like you have something else currently

Reply:
Sorry, my fault... It should now be what you expect.

But - it does look like something is failing, but I don't think it's related to my change?   Either that or something odd is up with Travis.  Mac failed, but linux is acting terribly slow.
See: https://travis-ci.org/github/tmaslach/vscode-cmake-tools

Reply:
It looks related to the line that filters out undefined results.  I think we need to remove the filter and replace with:
```ts
        const [, , quoted_name, unquoted_name, typename, valuestr] = match;
        const name = quoted_name || unquoted_name;
```

Reply:
Interesting.. Wasn't expecting that to be related.  

Alright, updated the change to include the lines you suggested.  Too bad, was trying hard not to have a quoted and unquoted variable! 

All is green!

Reply:
> Too bad, was trying hard not to have a quoted and unquoted variable!

No worries! 🤷‍♂ 😄